plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticator.java

```diff
@@ -133,7 +133,7 @@ private Principal authenticateWithUserBind(Credential credential)
                 log.debug("Authentication successful for user [%s]", user);
                 return new BasicPrincipal(user);
             }
-            catch (NamingException e) {
+            catch (NamingException | AccessDeniedException e) {
```

Comment:
Won't this also swallow actual `AccessDeniedException` if we run out of all patterns and hence violate the `PasswordAuthenticator` SPI?

Maybe we should check if the `lastException` is an `AccessDeniedException` before throwing a `RuntimeException`.

cc: @kokosing @lukasz-walkiewicz ?

Reply:
Thanks for uncovering a gap in our testing strategy - we fixed running related tests in https://github.com/trinodb/trino/pull/8125.

And thanks for the initial change - I created https://github.com/trinodb/trino/pull/8134 using your PR as a base.